File: Python/BOJ/BOJ16120.py
import sys
str = sys.stdin.readline().rstrip()
ppap = ["P", "P", "A", "P"]
stack = []
for char in str:
    stack.append(char)
    if stack[-4:] == ppap:
        stack.pop()
        stack.pop()
        stack.pop()
print("PPAP" if stack == ["P"] or stack == ppap else "NP")

File: swift/boj/boj16120.swift
import Foundation

let str = readLine()!.map { String($0) }
let ppap = ["P", "P", "A", "P"]
var stack: [String] = []
for char in str {
    stack.append(char)
    if Array(stack.suffix(4)) == ppap {
        stack.removeLast(3)
    }
}
print(stack.joined() == "P" || stack.joined() == "PPAP" ? "PPAP" : "NP")
